### PR TITLE
Feat/lifecycleState formR parts

### PR DESCRIPTION
**Pull request by hihilary**
1. add lifecycleState field in formRPart A & B
2. refactor GET FormRPart endpoints
3. add PUT FormRPart endpoints
4. unit tests

Please do NOT merge until the trainee-ui PR is raised.

### CODE REVIEW

File: src/main/java/uk/nhs/hee/tis/trainee/forms/api/FormRPartBResource.java [hunk 1]
```diff
@@ -72,17 +75,58 @@ public FormRPartBResource(FormRPartBService formRPartBService) {
   }
 
   /**
-   * GET /formr-partb/:traineeTisId.
+   * PUT /formr-partb : Update a FormRPartB.
+   *
+   * @param formRPartBDto the formRPartBDto to update
+   * @return the ResponseEntity with status 200 and with body the new formRPartBDto, or with status
+   * 500 (Internal Server Error) if the placementDTO couldn't be updated. If the id is not provided,
```
Comment (babul3012) [inline src/main/java/uk/nhs/hee/tis/trainee/forms/api/FormRPartBResource.java:82]: Same here.

File: src/main/java/uk/nhs/hee/tis/trainee/forms/mapper/FormRPartAMapper.java [hunk 2]
```diff
@@ -36,4 +37,8 @@
   List<FormRPartADto> toDtos(List<FormRPartA> formRPartAs);
```
Comment (babul3012) [inline src/main/java/uk/nhs/hee/tis/trainee/forms/mapper/FormRPartAMapper.java:37]: Is this method still used?

File: src/main/java/uk/nhs/hee/tis/trainee/forms/api/FormRPartAResource.java [hunk 3]
```diff
@@ -73,17 +76,58 @@ public FormRPartAResource(FormRPartAService formRPartAService) {
   }
 
   /**
-   * GET /formr-parta/:traineeTisId.
+   * PUT  /formr-parta : Update a FormRPartA.
+   *
+   * @param formRPartADto the formRPartADto to update
+   * @return the ResponseEntity with status 200 and with body the new formRPartADto, or
+   *     with status 500 (Internal Server Error) if the placementDTO couldn't be updated.
```
[the remark this answers is not in the document]
Comment (hihilary) [reply src/main/java/uk/nhs/hee/tis/trainee/forms/api/FormRPartAResource.java]: stupidly missed this, thanks Babul, it's fixed.

File: src/main/java/uk/nhs/hee/tis/trainee/forms/api/FormRPartBResource.java (hunk 1 above)
Comment (hihilary) [reply src/main/java/uk/nhs/hee/tis/trainee/forms/api/FormRPartBResource.java]: Fixed.

File: src/main/java/uk/nhs/hee/tis/trainee/forms/mapper/FormRPartAMapper.java (hunk 2 above)
Comment (hihilary) [reply src/main/java/uk/nhs/hee/tis/trainee/forms/mapper/FormRPartAMapper.java:37]: It's not used atm, but I think these methods are just some basic mappers, and can be left for later use.

**Pull request merged**